test/remote/device_tests.js

```diff
@@ -369,99 +369,6 @@ describe('remote device', function() {
-      var deviceInfo = {
-        name: 'test device',
-        type: 'mobile'
-      }
-      return Client.create(config.publicUrl, email, password, {
-        createdAt: '0'
-      })
-      .then(
-        function (client) {
-          return client.updateDevice(deviceInfo)
-            .then(
-              function () {
-                return client.devices()
-              }
-            )
-            .then(
-              function (devices) {
-                assert.equal(devices.length, 1, 'devices returned one item')
-                assert.strictEqual(devices[0].lastAccessTime, 0, 'devices returned correct lastAccessTime')
-                assert.strictEqual(devices[0].lastAccessTimeFormatted, '',
-                  'devices returned empty lastAccessTimeFormatted because lastAccesstime is 0')
-                assert.equal(devices[0].name, deviceInfo.name, 'devices returned correct name')
-                assert.equal(devices[0].type, deviceInfo.type, 'devices returned correct type')
-                return client.destroyDevice(devices[0].id)
-              }
-            )
-        }
-      )
-    }
-  )
-
-  it(
-    'devices list, sessionToken.lastAccessTime === -1',
-    () => {
-      var email = server.uniqueEmail()
-      var password = 'test password'
-      return Client.create(config.publicUrl, email, password, {
-        createdAt: '-1'
-      })
-      .then(
-        () => assert(false),
-        err => {
-          assert.equal(err.code, 400)
-        }
-      )
-
-    }
-  )
-
-  it(
-    'devices list, sessionToken.lastAccessTime === THE FUTURE',
-    () => {
-      var email = server.uniqueEmail()
-      var password = 'test password'
-      var deviceInfo = {
-        name: 'test device',
-        type: 'mobile'
-      }
-      var theFuture = Date.now() + 10000
-      return Client.create(config.publicUrl, email, password, {
-        createdAt: '' + theFuture
-      })
-      .then(
-        function (client) {
-          return client.updateDevice(deviceInfo)
-            .then(
-              function () {
-                return client.devices()
-              }
-            )
-            .then(
-              function (devices) {
-                assert.equal(devices.length, 1, 'devices returned one item')
-                assert.ok(devices[0].lastAccessTime > 0, 'devices returned correct lastAccessTime')
-                assert.ok(devices[0].lastAccessTime < theFuture, 'devices returned correct lastAccessTime')
-                assert.strictEqual(devices[0].lastAccessTimeFormatted, 'a few seconds ago',
-                  'devices returned correct lastAccessTimeFormatted')
-                assert.equal(devices[0].name, deviceInfo.name, 'devices returned correct name')
-                assert.equal(devices[0].type, deviceInfo.type, 'devices returned correct type')
-                return client.destroyDevice(devices[0].id)
-              }
-            )
-        }
-      )
-    }
-  )
-
```

Comment:
Ok, I've audited what these tests were doing:

- `sessionToken.lastAccessTime === 0`: Testing that the route allows returning a last access time of 0, because of the default value in the DB of 0. I've added a local unit test making asserting that `lastAccessTime: 0` passes the Joi validation.
- `sessionToken.lastAccessTime === -1`: Simply is testing that `/account/create` fails with `_createdAt=-1`. Since that parameter is not even accepted anymore, test is no longer needed.
- `sessionToken.lastAccessTime === FUTURE`: The access time is not adjusted anywhere in devices.upsert or devices.get routes, so this was only testing the initial creation of a device with a sessionToken from the future. The only code that adjusted the timestamp if it was in the future was in `optionallyOverrideCreatedAt`. So, the test was exercising our optional override, which has also been removed in this PR.

I believe the tests now accurately reflect the world.